pkg/templates/nstemplatetiers/nstemplatetier_generator_whitebox_test.go

```diff
@@ -182,76 +183,232 @@ func TestNewNSTemplateTier(t *testing.T) {
 			require.NoError(t, err)
 			namespace := "host-operator-" + uuid.NewV4().String()[:7]
 			// when
-			tierObjs, err := newNSTemplateTiers(s, namespace, templatesByTier)
+			tmplTiers, err := newNSTemplateTiers(decoder, namespace, templatesByTier)
 			// then
 			require.NoError(t, err)
-			require.NotEmpty(t, tierObjs)
+			require.NotEmpty(t, tmplTiers)
 			decoder := serializer.NewCodecFactory(s).UniversalDeserializer()
-			for _, actual := range tierObjs {
+			for _, actual := range tmplTiers {
 				tier := actual.Name
-				assert.Equal(t, namespace, actual.Namespace)
-				require.Len(t, actual.Spec.Namespaces, len(templatesByTier[tier].namespaceTemplates))
-				for kind, tmpl := range templatesByTier[tier].namespaceTemplates {
-					found := false
-					for _, ns := range actual.Spec.Namespaces {
-						if ns.Type != kind {
-							continue
+				t.Run(tier, func(t *testing.T) {
+					assert.Equal(t, namespace, actual.Namespace)
+					require.Len(t, actual.Spec.Namespaces, len(templatesByTier[tier].namespaceTemplates))
+					for kind, tmpl := range templatesByTier[tier].namespaceTemplates {
+						found := false
+						for _, ns := range actual.Spec.Namespaces {
+							if ns.Type != kind {
+								continue
+							}
+							found = true
```

Comment:
shouldn't we verify in this test also the references to the TierTemplates?

Reply:
done in 1cf9ca3